packages/marketplace/src/components/pages/swagger.tsx

```diff
@@ -10,35 +10,48 @@ import { StringMap } from '../../types/core'
 
 export const handleOnComplete = setLoading => () => setLoading(false)
 
-export const fetchInterceptor = async (params: StringMap) => {
+export const fetchAccessToken = async (setAccessToken: React.Dispatch<React.SetStateAction<null | string>>) => {
+  const fetchedAccessToken = await getAccessToken()
+  setAccessToken(fetchedAccessToken)
+}
+
+export const fetchInterceptor = (params: StringMap, accessToken: string | null) => {
   if (params.url === process.env.SWAGGER_BASE_URL) {
     return params
   }
+
   return {
     ...params,
     headers: {
       'Content-Type': 'application/json',
       'api-version': 'latest',
-      Authorization: `Bearer ${await getAccessToken()}`,
+      Authorization: `Bearer ${accessToken}`,
     },
   }
 }
 
 export const SwaggerPage: React.SFC = () => {
   const [loading, setLoading] = React.useState(true)
+  const [accessToken, setAccessToken] = React.useState()
+  const requestInterceptor = (params: StringMap) => fetchInterceptor(params, accessToken)
+
+  React.useEffect(() => {
+    fetchAccessToken(setAccessToken)
+  })
```

Comment:
a small thing, do we need `[fetchAccessToken]` as a dependency here to avoid `fetchAccessToken` from being called again and again?

Reply:
So fetchAccessToken calls getAccessToken which contains all the logic that checks if the token is valid / has expired etc.
It is not sufficient to check if the token exists (because it could exist but be invalid), so I call out to the method which performs the check on validity of the token and just returns it if it's valid so it's not expensive.
If it has expired however, it will refresh my session etc